generators/csharp/sdk/src/endpoint/http/RawClient.ts

```diff
@@ -85,6 +85,12 @@ export class RawClient {
                 assignment: csharp.codeblock(headerBagReference)
             });
         }
+        if (endpoint.requestBody && "contentType" in endpoint.requestBody && endpoint.requestBody.contentType) {
```

Comment:
We normally try to be more explicit when we resolve specific properties from discriminated unions. That way if we ever change the shape of the type in a later IR version, we know exactly where we need to change the code and we aren't ever surprised if we break Content-Type detection somewhere.

What are your thoughts on something like this instead?

```typescript
function getContentTypeFromRequestBody(requestBody: HttpRequestBody): string | undefined {
  switch (requestBody.type) {
    case "inlinedRequestBody":
      return requestBody.contentType;
    case "reference":
      return requestBody.contentType;
    case "fileUpload":
      return undefined;
    case "bytes":
      return requestBody.contentType;
    default:
      assertNever(requestBody);
  }
}
```